cfpb/cfgov-refresh
# Careers: add granular location data

Add the ability to associate granular location data (states and cities) with job regions and offices so that information can be added to job posting pages

## Additions

- City and state models in jobmanager
- Separate region and office models
- Ability to associate cities and states with regions and one city with an office

## Testing

1. Check out branch
2. Run `python cfgov/manage.py migrate`
3. Run `python cfgov/manage.py runscript create_career_locations`
4. Verify that cities and states can be added to job offices and regions in jobmanager admin
5. Publish a job posting page with a region location and verify that hidden city and state data is present
6. Publish a job posting page with an office location and verify that location data is displayed in page
7.  Publish a job posting page with an office location and remote option checked and verify that location data is displayed in page

## Screenshots
### Office admin
<img width="820" alt="screen shot 2018-01-25 at 9 26 24 am" src="https://user-images.githubusercontent.com/778171/35403169-beda3704-01b3-11e8-9804-fc250a5cdfa3.png">

### Region admin
![admin](https://user-images.githubusercontent.com/778171/35527764-ad33b41e-04e0-11e8-9047-109d47b0b1ab.png)

### Job posting with region location and hidden data
<img width="797" alt="screen shot 2018-02-02 at 2 15 07 pm" src="https://user-images.githubusercontent.com/778171/35757600-b0728f82-0824-11e8-942b-65b46cf1df4e.png">

### Job posting with office location
<img width="783" alt="screen shot 2018-01-25 at 9 36 12 am" src="https://user-images.githubusercontent.com/778171/35403237-fd22026c-01b3-11e8-995f-fcfeeb1e539f.png">

### Job posting with office location and remote option
<img width="747" alt="screen shot 2018-01-25 at 9 36 30 am" src="https://user-images.githubusercontent.com/778171/35403236-f97acaa4-01b3-11e8-8942-93c970dcc526.png">

## Checklist
